RFC: drivers: uart_ns16550: Improve macrobatic

finikorg opened the pull request:
Improve extensive macrobatic around io_map.

Make macro code more readbable. Basically factoring construction
like:

```
 #if MACRO
 if (...) {
     code
 } else {
 #else
        {
 #endif
```
to something like:
``` 
#if MACRO
 if (...) {
     code
     return;
 }
 #endif
```

jhedberg commented:
@finikorg I guess you should provide at least a little bit more info in the commit message. E.g. that this is trading off some memory (struct member always being present) for simpler code.

finikorg committed:
drivers: uart_ns16550: Improve macrobatic

Make macro code more readbable. Basically factoring construction
like:

 #if MACRO
 if (...) {
     code
 } else {
 #else
        {
 #endif

to something like:
 #if MACRO
 if (...) {
     code
     return;
 }
 #endif

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

dcpleung reviewed (commented):
IIRC, MISRA discourages return in the middle of function... and that return must be at the end.

MaureenHelm commented:
@finikorg do you plan to return to this PR?
